test/TestCases/SpanMemberFunctions.cpp

```diff
@@ -44,3 +44,38 @@ TEST_CASE ("Assignment operator", "[SpanMemberFunctions]")
     strings = std::move (mv);
     REQUIRE_THAT (strings, vctr::Equals ({ "I", "love", "sonible" }));
 }
+
+TEST_CASE ("castTo", "[SpanMemberFunctions]")
+{
+    SECTION ("Static extent")
+    {
+        constexpr vctr::Array<int32_t, 32> a { 0 };
+
+        auto asInt16Fixed = vctr::Span (a).castTo<const int16_t>();
+        auto asInt64Fixed = vctr::Span (a).castTo<const int64_t>();
+
+        STATIC_REQUIRE (asInt16Fixed.size() == 64);
+        STATIC_REQUIRE (asInt64Fixed.size() == 16);
+
+        REQUIRE (asInt16Fixed.allElementsEqual (0));
```

Comment:
Given that the original array has only 0 as members, I am not sure whether this actually tests anything substantial, except of no flipped bits are getting introduced.
Wouldn't it make more sense to initialise `a` to `1 << 16 | 1` and compare against `1` (int16 case) or `1 << 48  |  1 << 32 | 1 << 16 | 1` (64bit case)?

Reply:
Yes, that's a clever idea, will do that